internal/validate/validate.go

```diff
@@ -120,35 +217,47 @@ func evalVariables(compiler *cel.Compiler, vars map[string]interface{}, variable
 
 func evalValidations(compiler *cel.Compiler, vars map[string]interface{}, validations []settings.Validation) ([]byte, error) {
 	for _, validation := range validations {
-		ast, err := compiler.CompileCELExpression(validation.Expression)
+		message, reason, err := evaluateValidation(compiler, vars, validation)
 		if err != nil {
-			return nil, fmt.Errorf("failed to compile expression: %w", err)
+			return nil, err
 		}
 
-		val, err := compiler.EvalCELExpression(vars, ast)
-		if err != nil {
-			return nil, fmt.Errorf("failed to evaluate expression: %w", err)
+		if message != "" {
+			return kubewarden.RejectRequest(message, reason)
 		}
+	}
+	return kubewarden.AcceptRequest()
+}
 
-		if val == types.False {
-			reason := reasonToStatusCode(validation.Reason)
+func evaluateValidation(compiler *cel.Compiler, vars map[string]any, validation settings.Validation) (kubewarden.Message, kubewarden.Code, error) {
+	ast, err := compiler.CompileCELExpression(validation.Expression)
+	if err != nil {
+		return "", 0, fmt.Errorf("failed to compile expression: %w", err)
```

Comment:
Given than in evalValidations we only call the `kubewarden.RejectRequest()` or `kubewarden.AcceptRequest()`, we could as well just return it as part of this new `evaluateValidation()` instead of passing empty strings and 0 status codes, which to me feel more opaque.

Reply:
I realize now that `AcceptRequest()` and `RejectRequest()` are functions and there's no types for the requests, so we would end up passing `((kubewarden.Message, kubewarden.Code), error)` or a new type, which is too much work for this.

Thanks for cleaning up the function, resolving.